exasol/analytics/audit/audit.py

```diff
@@ -23,3 +38,113 @@ def __init__(
             name=TableNameImpl(table_name, SchemaName(db_schema)),
             columns=(BaseAuditColumns.all + additional_columns),
         )
+        self._column_names = [c.name for c in self.columns]
+
+    def augment(self, queries: Iterator[Query]) -> Iterator[str]:
+        for query in queries:
+            if not query.audit:
+                yield query.query_string
+            elif isinstance(query, AuditQuery):
+                yield self._insert(query)
+            elif isinstance(query, ModifyQuery):
+                yield from self._wrap(query)
+            else:
+                raise TypeError(
+                    f"Unexpected type {type(query)}" f" of query {query.query_string}"
+                )
+
+    def _insert(self, query: AuditQuery) -> str:
+        sub_query_alias = TableNameImpl("SUB_QUERY")
+        subquery_columns = {
+            c.name.name: ColumnName(c.name.name, sub_query_alias)
+            for c in output_columns(query.select_with_columns)
+        }
+        insert_statement = (
+            InsertStatement(self._column_names, separator=",\n  ")
+            .add_constants(query.audit_fields)
+            .add_scalar_functions(BaseAuditColumns.values)
+            .add_references(subquery_columns)
+        )
+
+        return (
+            f"INSERT INTO {self.name.fully_qualified} (\n"
+            f"  {insert_statement.columns}\n"
+            ") SELECT\n"
+            f"  {insert_statement.values}\n"
+            "FROM VALUES (1) CROSS JOIN\n"
```

Comment:
What if you don't have a subquery, I would push the cross join on the next line and surround it with a check

Or actually better why at all use values 1, remove it and the cross join and put the from after the check

Reply:
Agreed. I will fix this.

Reply:
See latest push - and thanks for your proposal making the implementation shorter, simpler, more elegant.